src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php

```diff
@@ -0,0 +1,779 @@
+{
+    /**
+     * The mysql tablename of the tree.
+     *
+     * @var string
+     */
+    private $treeTable = 'cms_tree';
+
+    /**
+     * @var string
+     */
+    private $rootNodeId = '';
+
+    /**
+     * @var string
+     */
+    private $currentPageId = '';
+
+    /**
+     * @var string
+     */
+    private $primaryConnectedNodeIdOfCurrentPage = '';
+
+    /**
+     * Nodes that should not be assignable or that should have only a
+     * restricted context menu.
+     *
+     * @var array
+     */
+    private $restrictedNodes = array();
+
+    /**
+     * Count of portals in this CMS
+     * tree level pre rendering is based on this.
+     *
+     * @var int
+     */
+    private $portalCount = 1;
+
+    /**
+     * @var Connection
+     */
+    private $dbConnection;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var InputFilterUtilInterface
+     */
+    private $inputFilterUtil;
+
+    /**
+     * @var BackendTreeNodeFactory
+     */
+    private $backendTreeNodeFactory;
+
+    /**
+     * @var PortalDomainServiceInterface
+     */
+    private $portalDomainService;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var UrlUtil
+     */
+    private $urlUtil;
+
+    /**
+     * @var NestedSetHelperFactoryInterface
+     */
+    private $nestedSetHelperFactory;
+
+    /**
+     * @var CacheInterface
+     */
+    private $cache;
+
+    public function __construct(
+        Connection $dbConnection,
+        EventDispatcherInterface $eventDispatcher,
+        InputFilterUtilInterface $inputFilterUtil,
+        BackendTreeNodeFactory $backendTreeNodeFactory,
+        PortalDomainServiceInterface $portalDomainService,
+        TranslatorInterface $translator,
+        UrlUtil $urlUtil,
+        NestedSetHelperFactoryInterface $nestedSetHelperFactory,
+        CacheInterface $cache
+    ) {
+        $this->dbConnection = $dbConnection;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->inputFilterUtil = $inputFilterUtil;
+        $this->backendTreeNodeFactory = $backendTreeNodeFactory;
+        $this->portalDomainService = $portalDomainService;
+        $this->translator = $translator;
+        $this->urlUtil = $urlUtil;
+        $this->nestedSetHelperFactory = $nestedSetHelperFactory;
+        $this->cache = $cache;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function Accept(\IMapperVisitorRestricted $visitor, $cachingEnabled, \IMapperCacheTriggerRestricted $cacheTriggerManager)
+    {
+        $isInIframe = $this->inputFilterUtil->getFilteredInput('isInIframe', '0');
+        $visitor->SetMappedValue('isInIframe', $isInIframe);
+
+        $noAssignDialog = $this->inputFilterUtil->getFilteredGetInput('noassign', '0');
+        $visitor->SetMappedValue('noAssignDialog', $noAssignDialog);
+
+        $tools = $this->getTools();
```

Comment:
Please add tools to the injected services instead.

Reply:
ok